src/lib.rs

```diff
@@ -18,7 +18,7 @@
 
 extern crate bytes;
 #[macro_use] extern crate futures;
```

Comment:
I think there's a recommendation that libraries shouldn't depend on `futures` directly, but rather on the individual crates (like `futures-core`, `futures-channel`, etc). The idea is that hyper could keep working if some extension needed a breaking change, and we only ever publicly claimed to depend on `futures-core`...

It'd be good to take inventory of what public dependencies we may be exposing here, either directly or indirectly:

- `futures-core` since things like `FutureResponse` implement `Future`.
- `futures-io` since we accept `AsyncRead` and `AsyncWrite`...
- I think all usage of `futures-channel` is no longer publicly exposed, so not a public dependency, right?
- While we use `FutureExt` internally, any types returned from it aren't viewable from outside the crate, I think.

Reply:
Good points! We have the same situation with `tokio`, where I was gonna fix that up in another PR.

Your call whether to do it as part of this PR, or whether I'll follow up with a combined `futures` and `tokio` PR for it.